earthaccess/store.py

```diff
@@ -342,20 +344,14 @@ def _open_granules(
                     print(f"using provider: {provider}")
                     s3_fs = self.get_s3fs_session(provider=provider)
             else:
-                access_method = "on_prem"
+                access = "on_prem"
                 s3_fs = None
 
-            data_links = list(
-                chain.from_iterable(
-                    granule.data_links(access=access_method) for granule in granules
-                )
-            )
-
+            url_mapping = _get_url_granule_mapping(granules, access)
```

Comment:
The basic idea here is we now construct this mapping from filename to granule that contains the file. Currently we just have a list of files and list of granules, which isn't quite what we need